src/priority/priority.py

```diff
@@ -295,6 +295,11 @@ def __init__(self, maximum_streams=1000):
         self._root_stream = Stream(stream_id=0, weight=1)
         self._root_stream.active = False
         self._streams = {0: self._root_stream}
+
+        if not isinstance(maximum_streams, int):
+            raise TypeError("maximum_streams must be an int.")
+        if maximum_streams <= 0:
```

Comment:
Just to confirm there's no use having zero streams as maximum. Otherwise LGTM :)